src/Native/System.Private.CoreLib.Native/pal_time.cpp

```diff
@@ -33,3 +34,161 @@ mach_timebase_info_data_t *InitializeTimebaseInfo()
     return g_isMachTimebaseInfoInitialized ? &g_machTimebaseInfo : nullptr;
 }
 #endif
+
+extern "C" uint64_t CoreLibNative_GetHighPrecisionCounts()
+{
+    uint64_t counts = 0;
+
+#if HAVE_MACH_ABSOLUTE_TIME
+    {
+        counts = mach_absolute_time();
+    }
+#elif HAVE_CLOCK_MONOTONIC_COARSE || HAVE_CLOCK_MONOTONIC
+    {
+        clockid_t clockType =
+#if HAVE_CLOCK_MONOTONIC_COARSE
+            CLOCK_MONOTONIC_COARSE; // good enough resolution, fastest speed
```

Comment:
GetHighPrecisionCount shouldn't use the coarse clock, and GetTickCount should try to use it if it's available